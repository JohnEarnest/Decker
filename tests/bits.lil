
# test suite for the bits interface.
show[bits]

# basic bitwise ops:
show[bits.and[12 10]]
show[bits.or [12 10]]

# bitwise ops conform:
show[bits.xor[15 (range 16)]]

# bitwise ops reduce:
show[bits.or[32 16 8 4 2 1]]

# bitwise ops reduce a single arg:
show[bits.or[(32,16,8,4,2,1)]]

# properly handle lengths 0 and 1:
show[bits.and[5]]
show[bits.and[()]~nil]
show[bits.and[]~nil]

# bounds
show[bits.or[2^8,16,24,30,31]]
