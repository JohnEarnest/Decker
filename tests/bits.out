<bits>
8
14
(15,14,13,12,11,10,9,8,7,6,5,4,3,2,1,0)
63
63
5
1
1
3238068480
